zio/csupio/stream.go

```diff
@@ -1,33 +1,60 @@
 package csupio
 
 import (
+	"context"
 	"io"
 	"math"
+	"runtime"
 	"sync"
 
 	"github.com/brimdata/super/csup"
 )
 
 type stream struct {
-	mu  sync.Mutex
-	r   io.ReaderAt
-	off int64
+	r    io.ReaderAt
+	ch   chan chunk
+	once sync.Once
+	ctx  context.Context
 }
 
-func (s *stream) next() (*csup.Object, error) {
+type chunk struct {
```

Comment:
Nit: We'd usually call a type that packages values for a channel like this "result" or "streamResult".